include/mbedtls/md.h

```diff
@@ -363,7 +366,7 @@ int mbedtls_md_file( const mbedtls_md_info_t *md_info, const char *path,
  *
  * \param ctx       The message digest context containing an embedded HMAC
  *                  context.
- * \param key       The HMAC secret key.
+ * \param key       The HMAC secret key. This must not be \c NULL.
```

Comment:
We should be more precise here. Suggestion: This must be a readable buffer of length \p keylen Bytes. It may be \c NULLL if \p keylen is zero.